File: server/src/models/Task.js
const mongoose = require("mongoose");

const taskSchema = new mongoose.Schema(
  {
    taskNumber: {
      type: Number,
      required: true,
      unique: true,
    },

    title: {
      type: String,
      required: true,
      trim: true,
      minlength: 1,
      maxlength: 120,
    },

    description: {
      type: String,
      required: false,
      trim: true,
    },

    status: {
      type: String,
      enum: ["Todo", "In Progress", "In Review", "Done"],
      default: "Todo",
    },

    tag: {
      type: String,
      trim: true,
      maxlength: 50,
      default: null,
    },

    assignee: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      default: null,
    },
  },
  {
    timestamps: true,
  },
);

module.exports = mongoose.model("Task", taskSchema);

File: mobile/app/src/main/java/com/devorbit/taskoff/data/models/Task.kt
package com.devorbit.taskoff.data.models

import com.google.gson.annotations.SerializedName

/**
 * Task Status Enum matching backend schema: ["Todo", "In Progress", "In Review", "Done"]
 */
enum class TaskStatus(val value: String) {
    @SerializedName("Todo")
    TODO("Todo"),

    @SerializedName("In Progress")
    IN_PROGRESS("In Progress"),

    @SerializedName("In Review")
    IN_REVIEW("In Review"),

    @SerializedName("Done")
    DONE("Done");

    companion object {
        fun fromString(statusStr: String?): TaskStatus {
            return entries.find { it.value.equals(statusStr, ignoreCase = true) } ?: TODO
        }
    }
}

/**
 * Task Data Model matching Backend Task schema & TECHNICAL_SPEC.md
 */
data class Task(
    @SerializedName("_id")
    val id: String,

    @SerializedName("taskNumber")
    val taskNumber: Int,

    @SerializedName("title")
    val title: String,

    @SerializedName("description")
    val description: String? = null,

    @SerializedName("status")
    val status: TaskStatus = TaskStatus.TODO,

    @SerializedName("tag")
    val tag: String? = null,

    @SerializedName("assignee")
    val assignee: User? = null,

    @SerializedName("createdAt")
    val createdAt: String? = null,

    @SerializedName("updatedAt")
    val updatedAt: String? = null
) {
    /**
     * Formatted task number for UI display (e.g., "#014")
     */
    val formattedTaskNumber: String
        get() = "#%03d".format(taskNumber)
}
